instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/GdprLogsE2ETests.kt

```diff
@@ -0,0 +1,168 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.nightly.logs
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import com.datadog.android.Datadog
+import com.datadog.android.log.Logger
+import com.datadog.android.nightly.utils.initializeSdk
+import com.datadog.android.nightly.utils.measureLoggerInitialize
+import com.datadog.android.nightly.utils.measureSdkInitialize
+import com.datadog.android.nightly.utils.measureSetTrackingConsent
+import com.datadog.android.nightly.utils.stopSdk
+import com.datadog.android.privacy.TrackingConsent
+import fr.xgouchet.elmyr.junit4.ForgeRule
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class GdprLogsE2ETests {
+
+    lateinit var logger: Logger
+
+    @get:Rule
+    val forge = ForgeRule()
+
+    // region Tests
+
+    @Before
+    fun setup() {
+        // The SDK is already started by default in the DatadogTestsRunner class so we need to
+        // stop it here to make sure we start it with the expected TrackingConsent in each test.
+        stopSdk()
```

Comment:
Maybe we should not initialize the SDK in the Runner after all as there will be plenty of cases where we need to configure the SDK initialization in the test. I think I am going to remove that code from the Runner and expect each test class to initialize the SDK either in the `setUp` method or in the test case and to `stop` in the `tearDown` method. For the latter probably I will add a test rule to reinforce this behaviour. What do you think @xgouchet @0xnm

Reply:
Actually I think I can stop the SDK in the runner directly after each test.

Reply:
Yes let's have a rule to enforce stopping the SDK after each test, and have a utility method to initialise the SDK to at least share the Credentials

Reply:
After doing some several tests I realized we need to keep it this way as the SDK needs to run before the process ends otherwise there will be unsent events. Stoping it after each test will cancel our uploader and will not send all the events. For scenarios where we need to initialize the SDK in the test method we will have to follow the current approach and `stop` the current sdk in the `setUp` method in order to be able to start it again with the required configuration.